File: framework/data-jpa/src/test/java/top/bettercode/simpleframework/data/test/domain/Employee.java
package top.bettercode.simpleframework.data.test.domain;

import javax.persistence.EmbeddedId;
import javax.persistence.Entity;
import org.hibernate.annotations.ColumnDefault;
import org.hibernate.annotations.DynamicUpdate;
import org.hibernate.annotations.Type;
import top.bettercode.lang.util.StringUtil;
import top.bettercode.simpleframework.data.jpa.SoftDelete;

@DynamicUpdate
@Entity
public class Employee {

  @EmbeddedId
  private EmployeeKey employeeKey;
  private String firstName;
  private String lastName;
  @SoftDelete
  @Type(type = "org.hibernate.type.NumericBooleanType")
  @ColumnDefault("0")
  private Boolean deleted;

  public Employee() {
  }

  public Employee(EmployeeKey employeeKey, String firstName, String lastName) {
    this.employeeKey = employeeKey;
    this.firstName = firstName;
    this.lastName = lastName;
  }

  public EmployeeKey getEmployeeKey() {
    return employeeKey;
  }

  public void setEmployeeKey(EmployeeKey employeeKey) {
    this.employeeKey = employeeKey;
  }

  public String getFirstName() {
    return firstName;
  }

  public void setFirstName(String firstName) {
    this.firstName = firstName;
  }

  public String getLastName() {
    return lastName;
  }

  public void setLastName(String lastName) {
    this.lastName = lastName;
  }

  public Boolean getDeleted() {
    return deleted;
  }

  public void setDeleted(Boolean deleted) {
    this.deleted = deleted;
  }

  @Override
  public String toString() {
    return StringUtil.json(this);
  }
}

File: data-jpa/src/test/kotlin/top/bettercode/summer/data/jpa/domain/Employee.kt
package top.bettercode.summer.data.jpa.domain

import jakarta.persistence.EmbeddedId
import jakarta.persistence.Entity
import org.hibernate.annotations.ColumnDefault
import org.hibernate.annotations.DynamicUpdate
import org.hibernate.annotations.JdbcType
import org.hibernate.type.descriptor.jdbc.TinyIntJdbcType
import top.bettercode.summer.data.jpa.LogicalDelete
import top.bettercode.summer.tools.lang.util.StringUtil.json

@Suppress("LeakingThis")
@DynamicUpdate
@Entity
open class Employee {
    @EmbeddedId
    open var employeeKey: EmployeeKey? = null
    open var firstName: String? = null
    open var lastName: String? = null

    @LogicalDelete
    @JdbcType(TinyIntJdbcType::class)
    @ColumnDefault("0")
    open var deleted: Boolean? = null

    constructor()
    constructor(employeeKey: EmployeeKey?, firstName: String?, lastName: String?) {
        this.employeeKey = employeeKey
        this.firstName = firstName
        this.lastName = lastName
    }

    override fun toString(): String {
        return json(this)
    }
}